src/metabase/lib/fe_util.cljc

```diff
@@ -13,7 +13,9 @@
    [metabase.lib.temporal-bucket :as lib.temporal-bucket]
    [metabase.lib.util :as lib.util]
    [metabase.mbql.util :as mbql.u]
+   [metabase.shared.formatting.internal.date-builder :as shared.formatting.internal.builder]
```

Comment:
Would it be ok to use `metabase.shared.formatting.internal.date-builder` in this namespace?

Reply:
No, the point of "internal" is to disallow this kind of thing.

See below about new functions in `metabase.shared.formatting.date` that give you the strings you need behind a *much* simpler interface. The `date-builder` stuff should remain an implementation detail.